### PR TITLE
add private or public for create repo

### PR DESCRIPTION
**Pull request by dislazy**
add private or public for create repo

### CODE REVIEW

**Comment by Yikun** [comment]
Thanks for you contributions!

But in my opinion, this kind of breaks the semantics of synchronization. Could you share your real case? Let's see why we need this parameter. I guess you just want backup your repos but don't want dispaly these repos in your dst repo?

**Comment by Yikun** [comment]
also cc @yi-Xu-0100 @ShixiangWang @JasonkayZK What do you think?

**Comment by ShixiangWang** [comment]
@Yikun Thanks for cc me. Currently I don't have such demands. From the file change, it seems that it would not change default behavior. I agree with you that if a real case can be estimated, it's worthy to have this new option.

File: README_en.md [hunk 1]
```diff
@@ -47,6 +47,7 @@ More than [100+](https://github.com/search?p=2&q=hub-mirror-action+%22account_ty
 - `force_update` (optional) Force to update the destination repo, use '-f' flag do 'git push'
 - `timeout` (optional) Default is '30m', set the timeout for every git command, like '600'=>600s, '30m'=>30 mins, '1h'=>1 hours
 - `mappings` (optional) Default is empty, the source repos mappings, such as 'A=>B, C=>CC', source repo name would be mapped follow the rule: A to B, C to CC. Mapping is not transitive.
+- `dst_private` (optional) Default is false. After configuration, if the gitee repository does not exist, create a public repository, otherwise create a private repository
```
Comment (Yikun) [inline README_en.md:50]: same, hub-mirror is not only for gitee as dst.

**Comment by yi-Xu-0100** [comment]
@Yikun Thanks for cc me. If it can be used to release sync the private repo, it will close #38. That will be very nice. 😊

**Comment by dislazy** [comment]
Hello everyone, my main purpose is to achieve mutual backup between github and gitee. If the github repository is private, then if the repository synchronized by gitee is not private, some key information will be leaked.

File: hub-mirror/hub.py [hunk 2]
```diff
@@ -62,7 +63,7 @@ def create_dst_repo(self, repo_name):
         if self.dst_type == 'gitee':
             data = {'name': repo_name}
         elif self.dst_type == 'github':
-            data = json.dumps({'name': repo_name})
+            data = json.dumps({'name': repo_name,'private': self.dst_private})
         if not self.has_dst_repo(repo_name):
```
[the remark this answers is not in the document]
Comment (dislazy) [reply hub-mirror/hub.py:67]: I don't have similar problems, I now have several private repositories that are github synced to gitee's private repositories

**Comment by dislazy** [comment]
As such, yes, but you can continue to optimize

**Comment by Yikun** [comment]
See related commentes in here: https://github.com/Yikun/hub-mirror-action/issues/38#issuecomment-1094033841

Could change `dst_private` to `dst_visibility`?

File: README_en.md (hunk 1 above)
Comment (Yikun) [inline README_en.md:50]:
```suggestion
- `dst_private` (optional) Default is public. If set to `private`, new creating repo visibility is `private`, but if destination repo have been created, the repo visibility will not be set, just only update the repo.

File: hub-mirror/hub.py [hunk 3]
```diff
@@ -7,7 +7,7 @@
 
 class Hub(object):
     def __init__(
-        self, src, dst, dst_token, account_type="user",
+        self, src, dst, dst_token,dst_private, account_type="user",
```
Comment (Yikun) [inline hub-mirror/hub.py:10]: pep8 failed

**Comment by Yikun** [comment]
I added the comments according latest desigin.

**Comment by xiexianbin** [comment]
ping

**Pull request closed without merging**